chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/GetAllActionPermissions.java

```diff
@@ -0,0 +1,68 @@
+package gov.healthit.chpl.permissions.domains.scheduler;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.auth.Util;
+import gov.healthit.chpl.auth.permission.GrantedPermission;
+import gov.healthit.chpl.domain.schedule.ChplJob;
+import gov.healthit.chpl.permissions.domains.ActionPermissions;
+
+@Component(value = "schedulerGetAllActionPermissions")
+public class GetAllActionPermissions extends ActionPermissions {
+    private static final Logger LOGGER = LogManager.getLogger(GetAllActionPermissions.class);
+    private static final String AUTHORITY_DELIMITER = ";";
+
+    @Override
+    public boolean hasAccess() {
+        return Util.isUserRoleAdmin() || Util.isUserRoleOnc() || Util.isUserRoleAcbAdmin();
+    }
+
+    @Override
+    public boolean hasAccess(Object obj) {
+        try {
+            if (!(obj instanceof ChplJob)) {
+                return false;
+            } else if (Util.isUserRoleAcbAdmin() || Util.isUserRoleOnc()) {
+                ChplJob job = (ChplJob) obj;
+                return job.getGroup().equals("chplJobs") && doesUserHavePermissionToJob(job);
+            } else {
+                return Util.isUserRoleAdmin();
+            }
+        } catch (Exception e) {
+            LOGGER.error(e);
+            return false;
+        }
+    }
+
+    private Boolean doesUserHavePermissionToJob(final ChplJob job) {
+        //Get the authorities from the job
+        if (job.getGroup() != null && job.getGroup().equals("chplJobs")) {
```

Comment:
This method is only called from the `hasAccess` method, right? And from that method, only after we've already determined that the group == "chplJobs", so this if isn't needed, correct?

Reply:
True - change made.